src/lib.rs

```diff
@@ -293,15 +342,22 @@ impl<K: Eq + Clone> Sources<K> {
 
     /// Wait for readiness events on the given list of sources. If no event
     /// is returned within the given timeout, returns an error of kind `io::ErrorKind::TimedOut`.
-    pub fn wait_timeout(
-        &mut self,
-        events: &mut Events<K>,
-        timeout: time::Duration,
-    ) -> Result<(), io::Error> {
+    pub fn poll(&mut self, events: &mut Events<K>, timeout: Timeout) -> Result<(), io::Error> {
         events.initialize(&self);
 
-        let timeout = timeout.as_millis() as libc::c_int;
-        let result = self.poll(events, timeout);
+        let timeout = match timeout {
+            Timeout::After(duration) => duration.as_millis() as libc::c_int,
```

Comment:
I would probably have used `.try_into().unwrap()` here, but I note that you prefer not to panic.

I considered implementing `Timeout` using an `i32` internally instead of a `time::Duration` so that users would be prevented from specifying an invalid timeout, but that has its own problems, since it can be negative.

I could add bounds checking to `Timeout`, but that would involve adding an error type (for `TryFrom`), and possibly some panics in convenience methods (since `Timeout::from_secs(5).unwrap()` seems silly).

Reply:
Yeah I think it's fine like this. The clipping is not a new problem, so I don't consider this a regression. Additionally, the worse case scenario is a user wanted a longer than `i32::MAX` timeout, and it clips to `i32::MAX`, which is covered by the docs anywa.